web/pgadmin/tools/sqleditor/__init__.py

```diff
@@ -2098,7 +2098,15 @@ def start_query_download_tool(trans_id):
         )
 
     try:
-
+        sql = None
+        # Iterate through CombinedMultiDict to find query.
+        for key, value in data.items():
+            if key == 'query':
+                sql = value
+        if not sql:
+            sql = trans_obj.get_sql(sync_conn)
+        # Re-execute the query to ensure the latest data is included
+        sync_conn.execute_async(sql)
```

Comment:
what if there is no change in the data? In many cases data is not even editable. This is blind execute.